src/scripts/main.js

```diff
@@ -1,3 +1,54 @@
+/* eslint-disable no-unused-expressions */
 'use strict';
 
-// write code here
+const tableHead = document.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+
+tableHead.addEventListener('click', (e) => {
+  const columnName = e.target.closest('th');
+  const rows = tableBody.rows;
+  const people = [];
+
+  for (const row of rows) {
+    people.push({
+      Name: row.cells[0].textContent,
+      Position: row.cells[1].textContent,
+      Age: row.cells[2].textContent,
+      Salary: row.cells[3].textContent,
+    });
+  }
+
+  const sortedName = columnName.textContent.trim();
+
+  people.sort((person1, person2) => {
+    if (sortedName === 'Name' || sortedName === 'Position') {
+      return person1[sortedName].localeCompare(person2[sortedName]);
+    }
+
+    if (columnName.textContent === 'Age') {
+      return parseInt(person1[sortedName]) - parseInt(person2[sortedName]);
+    }
+
+    if (sortedName === 'Salary') {
+      const salary1 = parseInt(person1[sortedName].replace(/[^0-9]/g, ''), 10);
+      const salary2 = parseInt(person2[sortedName].replace(/[^0-9]/g, ''), 10);
+
+      return salary1 - salary2;
+    }
+  });
```

Comment:
The sort function does not handle cases where `sortedName` does not match any of the expected column names ('Name', 'Position', 'Age', 'Salary'). Consider adding a default case or handling unexpected column names to avoid potential errors.